File: backend/models/match.py
from pymongo import MongoClient
from bson.objectid import ObjectId
import os
from datetime import datetime

# MongoDB connection
client = MongoClient(os.getenv("MONGODB_URI"))
db = client[os.getenv("MONGODB_DB_NAME", "sportmatch")]

class Match:
    @staticmethod
    def get_by_user_id(user_id):
        return list(db.matches.find({
            "$or": [{"user1Id": user_id}, {"user2Id": user_id}]
        }))
    
    @staticmethod
    def get_by_id(id):
        if not ObjectId.is_valid(id):
            return None
        return db.matches.find_one({"_id": ObjectId(id)})
    
    @staticmethod
    def create(match_data):
        match_data["createdAt"] = datetime.now()
        match_data["updatedAt"] = datetime.now()
        
        result = db.matches.insert_one(match_data)
        match_data["_id"] = result.inserted_id
        
        return match_data
    
    @staticmethod
    def update(id, match_data):
        if not ObjectId.is_valid(id):
            return None
            
        match_data["updatedAt"] = datetime.now()
        
        result = db.matches.update_one(
            {"_id": ObjectId(id)},
            {"$set": match_data}
        )
        
        if result.modified_count > 0:
            return db.matches.find_one({"_id": ObjectId(id)})
        
        return None
    
    @staticmethod
    def delete(id):
        if not ObjectId.is_valid(id):
            return False
            
        result = db.matches.delete_one({"_id": ObjectId(id)})
        return result.deleted_count > 0

File: models/Match.ts
import { ObjectId } from "mongodb"
import clientPromise from "@/lib/mongodb"

export interface Match {
  _id?: ObjectId
  user1Id: string
  user2Id: string
  timestamp: Date
  hasChat: boolean
  createdAt?: Date
  updatedAt?: Date
}

export async function getMatchesByUserId(userId: string): Promise<Match[]> {
  try {
    const client = await clientPromise
    const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

    return db
      .collection("matches")
      .find({
        $or: [{ user1Id: userId }, { user2Id: userId }],
      })
      .sort({ createdAt: -1 })
      .toArray() as Promise<Match[]>
  } catch (error) {
    console.error("Error in getMatchesByUserId:", error)
    return []
  }
}

export async function createMatch(matchData: Omit<Match, "_id" | "createdAt" | "updatedAt">): Promise<Match> {
  const client = await clientPromise
  const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

  const now = new Date()
  const match = {
    ...matchData,
    createdAt: now,
    updatedAt: now,
  }

  const result = await db.collection("matches").insertOne(match)

  return {
    ...match,
    _id: result.insertedId,
  }
}

export async function getMatchById(id: string): Promise<Match | null> {
  try {
    const client = await clientPromise
    const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

    if (ObjectId.isValid(id)) {
      return db.collection("matches").findOne({ _id: new ObjectId(id) }) as Promise<Match | null>
    }

    return null
  } catch (error) {
    console.error("Error in getMatchById:", error)
    return null
  }
}
